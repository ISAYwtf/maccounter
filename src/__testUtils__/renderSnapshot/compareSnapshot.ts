import { render } from '@testing-library/react'
import { ReactElement } from 'react'

export const compareSnapshot = (component: ReactElement) => {
    const { asFragment } = render(component)
    expect(asFragment()).toMatchSnapshot()
}
