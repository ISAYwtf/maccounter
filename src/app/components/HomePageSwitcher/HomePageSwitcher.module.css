.switcher {
    display: flex;
    align-items: center;
}

.switcher p {
    margin-right: 25px;
    font-size: 24px;
}
