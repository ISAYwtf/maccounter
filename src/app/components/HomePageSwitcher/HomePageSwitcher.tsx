import React from 'react'
import Switcher from 'app/components/Switcher/Switcher'
import styles from './HomePageSwitcher.module.css'

const HomePageSwitcher: React.FC = () => (
    <div className={styles.switcher}>
        <p>Light mode</p>
        <Switcher />
    </div>
)

export default HomePageSwitcher
