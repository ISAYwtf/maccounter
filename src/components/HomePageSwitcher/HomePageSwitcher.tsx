import React from "react"
import styles from "./HomePageSwitcher.module.css"
import Switcher from "../Switcher/Switcher";

const HomePageSwitcher = () =>
    <div className={styles.switcher}>
        <p>Light mode</p>
        <Switcher />
    </div>

export default HomePageSwitcher
