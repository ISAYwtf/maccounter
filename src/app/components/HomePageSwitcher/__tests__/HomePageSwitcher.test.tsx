import React from 'react'
import HomePageSwitcher from '@components/HomePageSwitcher/HomePageSwitcher'
import { compareSnapshot } from '@testUtils/renderSnapshot'

describe('HomePageSwitcher', () => {
    const renderComponent = <HomePageSwitcher />

    // eslint-disable-next-line
    it('render HomePageSwitcher', () => compareSnapshot(renderComponent))
})
