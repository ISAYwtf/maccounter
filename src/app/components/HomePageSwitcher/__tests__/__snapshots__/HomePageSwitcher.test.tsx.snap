// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageSwitcher render HomePageSwitcher 1`] = `
<DocumentFragment>
  <div
    class="switcher"
  >
    <p>
      Light mode
    </p>
    <label
      class="switch"
    >
      <input
        type="checkbox"
      />
      <span
        class="slider"
      />
    </label>
  </div>
</DocumentFragment>
`;
